### DISCUSSION TITLE
Custom token props

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#42147] [General]
Hello, is there any way one can extend the `token` property in `ThemeConfig` so we can pass custom properties in the Token context?

**Comment by bombillazo** [comment]
If it helps anyone in the future, here is how I solved it:

```ts
// external.d.ts
`declare module 'antd/es/theme/internal' {
  /**
   * We can add custom props to the theme token here for use in our app.
   */
  interface AliasToken {
    colorPositive: string;
    colorNegative: string;
  }
}
```

**Reply by rafalm1** [reply]
Thank you, it works

**Reply by abdelkadir-bouziane** [reply]
> If it helps anyone in the future, here is how I solved it:
> 
> ```ts
> // external.d.ts
> `declare module 'antd/es/theme/internal' {
>   /**
>    * We can add custom props to the theme token here for use in our app.
>    */
>   interface AliasToken {
>     colorPositive: string;
>     colorNegative: string;
>   }
> }
> ```

Hello Mr. @bombillazo, can you please tell us more about the solution you've reached?
